feat(panel): Add the "Report a broken page" view

smalluban opened the pull request:
The PR adds the CTA for paused domains, which opens the "Report a broken page" in the panel. 

* It looks like `activeTab` permission does not trigger a browser popup when the extension is updated. However, we can also try out the optional permission if it is better.
* We need to match operating systems between the extension and support form - I think we should update fields to follow extension logic - `mac`, `win`, `ios`, `ipados`,`linux`, etc... and version only as a separate field (currently we have `windows_10` and other as os name)
* Copy is a draft version - needs review and improvements
* I tested the screenshot feature on Chrome, Firefox, and Safari on Mac and iOS, and it works as expected
* Open question about the link "Report a broken page" in the menu - should it also open the form in the panel or should it open the website as it does now?

WARNING: the PR logic is finished and it's connected to the real end-point. Please use `npm start -- --staging` when testing to avoid spaming the support channel.

chrmod commented on src/background/broken-page-report.js:
```diff
@@ -0,0 +1,67 @@
+          'support_ticket[subject]',
+          `[GBE] Broken page report: ${msg.url}`,
+        );
+        formData.append('support_ticket[message]', msg.description);
```
We should pass the product version via the form. Until it is supported by the website lets add it to the message. 

Lets consider if we want to add also: 
* list of detected activities (no urls) 
* extension settings 
* block list version (checksums)

chrmod commented on src/background/broken-page-report.js:
```diff
@@ -0,0 +1,67 @@
+        if (msg.screenshot) {
+          const screenshot = await chrome.tabs.captureVisibleTab(null, {
+            format: 'jpeg',
+            quality: 100,
```
currently the height defaults to the view port height, can we pass a height of the full page to capture everything?

smalluban replied:
The `rect` option is only supported in Firefox (https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/extensionTypes/ImageDetails#rect), and I think it might be tricky to detect the height of the whole page, and still I am not sure if without manual "scrolling" you can take a screenshot of the part, which is not visible. I would keep it as it is for now.

chrmod commented on src/background/broken-page-report.js:
```diff
@@ -0,0 +1,67 @@
+          }
+        });
+
+        sendResponse();
```
should we add some error handling so in case of a problem we let people know they can report the problem via the website?

smalluban replied:
Good point. That's technically easy, I will add a message after error handling, like "Error... However, you can always report the broken page via our [website form]"

chrmod reviewed (commented):
Looks great so far!

smalluban commented:
*  Updated copy from @mdsandu
* Added version, options and trackers to description
* Additional info when error occurs

I think the PR is ready for final review for merging.

smalluban commented:
There is a last open question - what about the link to report a broken page in the menu section of the panel? I am unsure if it should be changed, as the user might want to report "some" page - it does not have to be the one it is now... And if we change that, there will be no direct link to the form on the ghostery.com

mdsandu commented:
pls post current screens to resolve the last issue. thanks

smalluban commented:
Updated copy:

<img width="345" alt="Zrzut ekranu 2025-01-14 o 09 24 57" src="https://github.com/user-attachments/assets/0970a816-0340-4e64-8770-82cbdad349bc" />
<img width="344" alt="Zrzut ekranu 2025-01-14 o 09 29 48" src="https://github.com/user-attachments/assets/ecdf7858-9f7d-49ca-a065-8da66673cec6" />
<img width="348" alt="Zrzut ekranu 2025-01-14 o 09 30 35" src="https://github.com/user-attachments/assets/91baeda0-8ef2-4b28-9353-d88ba93306d7" />

Dimi-GE commented:
Screenshots from Zendesk:

![image](https://github.com/user-attachments/assets/9d6e4528-b645-4023-8ffc-2ae1cf0abc0b)

![image](https://github.com/user-attachments/assets/ecf6ad24-b170-4d84-a787-48e5a6f9c7f3)

![image](https://github.com/user-attachments/assets/ad9cecf7-6356-4868-a835-37cb56d42bc4)

![image](https://github.com/user-attachments/assets/04b2866a-a9d4-49ad-bf27-2134e98076c5)

mdsandu commented:
@smalluban two suggested changes to cover your concern
same text in 2 locations clarifying (hopefully)

- [ ] on screen 1 add "If the issue persists or you’d like to report a different page as broken, please use the report form on ghostery.com" below "Include a screenshot of the current page" with a bit of spacing in between. 


- [ ] on screen 2/fail screen: expand the red text to same message "If the issue persists or you’d like to report a different page as broken, please use the report form on ghostery.com"

let's see how this feels?

smalluban commented:
Keeping the message below is a good idea, as a case with an error might be rare. However, I don't like displaying it twice and hiding the second one (Ui flickering). I would keep only point 1 - display the message all the time regardless of the error (which can never be shown).

<img width="343" alt="Zrzut ekranu 2025-01-16 o 08 01 34" src="https://github.com/user-attachments/assets/3da59747-2c41-4273-89fe-2f102db5fec0" />
<img width="342" alt="Zrzut ekranu 2025-01-16 o 08 04 03" src="https://github.com/user-attachments/assets/d5ebd42a-58dd-43bf-9dc2-4ac9a384ca62" />

mdsandu commented:
hi @smalluban, not sure I fully understand the final state you are suggesting
how will the fail state look like (I am not a fan of showing the same message twice on a screen)
let's clarify offline

smalluban commented:
> I would keep only point 1 

It's simple - just keep the message below the "include screenshot" all the time, and not display another (or switch) when error happens.

mdsandu commented:
ah ok, now I get it! great! let's move on!

smalluban commented:
I tested Chrome, and there is no confirm popup after update with added `activeTab` permission. Using the same technique I am able to block the extension when adding `geolocation` permission (to be sure that I test it right).

I assume that only Chrome is affected by the "blocking popup" when permissions are added.
